docker-compose.yml

```diff
@@ -1,10 +1,9 @@
-version: '3.8'
 services:
   pingvin-share:
     image: stonith404/pingvin-share
     restart: unless-stopped
     ports:
-      - 3000:3000
+      - 3333:3333
```

Comment:
```suggestion
      - 3000:3000
```

Caddy (a reverse proxy) runs on port 3000. Because we have a backend and frontend that run on different ports, we need a reverse proxy. The frontend has a built-in reverse proxy, which is why the application is also able to run over port 3333, but the performance of the built-in reverse proxy is poor for file uploads.

Because of that, we should use port 3000.

Reply:
Then there has to be something wrong with it, because I couldn't get a clean container (v0.26.0) to work with port 3000.
I had to use 3333, else I wasn't able to access pingvin

Reply:
Oh, really? Can you see any errors in the logs, or is the connection just refused?

Reply:
No logs about an incoming request. My browser just gets `ERR_EMPTY_RESPONSE`
![image](https://github.com/stonith404/pingvin-share/assets/91284394/0c459974-9d23-4849-96dc-754893752dcf)

Reply:
Can you share the output of `docker compose exec pingvin-share curl 127.0.0.1:2019/config/`? And are you sure that you have the latest image of the `latest` tag?

Reply:
Yes, I did explicitly use `image: stonith404/pingvin-share:v0.26.0`.
```bash
docker exec 088fbbe90953 curl 127.0.0.1:2019/config/
curl: (7) Failed to connect to 127.0.0.1 port 2019 after 0 ms: Couldn't connect to server
```
In all honesty, it could very well just be my setup that doesn't work. This is the second time I report something not working, and I can't really believe it's only happening to me

I did actually set up a pingvin share on my work pc, just for testing, and I was able to reach `:3000` by default (while using the exact same compose file) iirc.
I can share the logs of that as well, but that'll have to wait until Monday ^^"

Reply:
Thanks for trying this on your work pc. It's strange because it seems like Caddy isn't running in your container. Can you share the whole logs of your container that isn't working?

Reply:
Sure, these are my local logs when starting the container:
```
G:\pingvin-discord>docker compose up
[+] Running 1/0
 ✔ Container pingvin-discord-pingvin-1  Created                                                                    0.0s
Attaching to pingvin-discord-pingvin-1
pingvin-discord-pingvin-1  | cp: can't create '/opt/app/frontend/public/img/favicon.ico': No such file or directory
pingvin-discord-pingvin-1  | cp: can't create directory '/opt/app/frontend/public/img/icons': No such file or directory
pingvin-discord-pingvin-1  | cp: can't create '/opt/app/frontend/public/img/logo.png': No such file or directory
pingvin-discord-pingvin-1  |   ▲ Next.js 14.2.3
pingvin-discord-pingvin-1  |   - Local:        http://localhost:3333
pingvin-discord-pingvin-1  |   - Network:      http://0.0.0.0:3333
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  |  ✓ Starting...
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  | > pingvin-share-backend@0.26.0 prod
pingvin-discord-pingvin-1  | > prisma migrate deploy && prisma db seed && node dist/src/main
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  | Environment variables loaded from prisma/.env
pingvin-discord-pingvin-1  | Prisma schema loaded from prisma/schema.prisma
pingvin-discord-pingvin-1  | Datasource "db": SQLite database "pingvin-share.db" at "file:../data/pingvin-share.db"
pingvin-discord-pingvin-1  |  ✓ Ready in 1886ms
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  | 18 migrations found in prisma/migrations
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  | No pending migrations to apply.
pingvin-discord-pingvin-1  | Environment variables loaded from prisma/.env
pingvin-discord-pingvin-1  | Running seed command `ts-node prisma/seed/config.seed.ts` ...
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  | 🌱  The seed command has been executed.
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  | (process:31): VIPS-WARNING **: 15:06:15.804: threads clipped to 1024
pingvin-discord-pingvin-1  |
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:15 PM     LOG [NestFactory] Starting Nest application...
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] PrismaModule dependencies initialized +117ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] JwtModule dependencies initialized +7ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] JwtModule dependencies initialized +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] ThrottlerModule dependencies initialized +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] DiscoveryModule dependencies initialized +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] CacheModule dependencies initialized +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] ScheduleModule dependencies initialized +0ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] AppModule dependencies initialized +1ms
pingvin-discord-pingvin-1  | Connected to the database
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] EmailModule dependencies initialized +80ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] ClamScanModule dependencies initialized +4ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] JobsModule dependencies initialized +4ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] ConfigModule dependencies initialized +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] FileModule dependencies initialized +8ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] UserModule dependencies initialized +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] ReverseShareModule dependencies initialized +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] AuthModule dependencies initialized +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] ShareModule dependencies initialized +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [InstanceLoader] OAuthModule dependencies initialized +0ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RoutesResolver] AppController {/api}: +52ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/health, GET} route +10ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RoutesResolver] AuthController {/api/auth}: +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/auth/signUp, POST} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/auth/signIn, POST} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/auth/signIn/totp, POST} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/auth/resetPassword/:email, POST} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/auth/resetPassword, POST} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/auth/password, PATCH} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/auth/token, POST} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/auth/signOut, POST} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/auth/totp/enable, POST} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/auth/totp/verify, POST} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/auth/totp/disable, POST} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RoutesResolver] ShareController {/api/shares}: +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/all, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares, GET} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/:id, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/:id/from-owner, GET} route +0ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/:id/metaData, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares, POST} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/:id/complete, POST} route +0ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/:id/complete, DELETE} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/:id, DELETE} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/isShareIdAvailable/:id, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/:id/token, POST} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RoutesResolver] FileController {/api/shares/:shareId/files}: +0ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/:shareId/files, POST} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/:shareId/files/zip, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/:shareId/files/:fileId, GET} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/shares/:shareId/files/:fileId, DELETE} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RoutesResolver] ReverseShareController {/api/reverseShares}: +0ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/reverseShares, POST} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/reverseShares/:reverseShareToken, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/reverseShares, GET} route +0ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/reverseShares/:reverseShareId, DELETE} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RoutesResolver] ConfigController {/api/configs}: +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/configs, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/configs/admin/:category, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/configs/admin, PATCH} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/configs/admin/testEmail, POST} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/configs/admin/logo, POST} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RoutesResolver] UserController {/api/users}: +0ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/users/me, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/users/me, PATCH} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/users/me, DELETE} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/users, GET} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/users, POST} route +0ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/users/:id, PATCH} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/users/:id, DELETE} route +0ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RoutesResolver] OAuthController {/api/oauth}: +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/oauth/available, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/oauth/status, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/oauth/auth/:provider, GET} route +2ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/oauth/callback/:provider, GET} route +1ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [RouterExplorer] Mapped {/api/oauth/unlink/:provider, POST} route +0ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:16 PM     LOG [NestApplication] Nest application successfully started +69ms
pingvin-discord-pingvin-1  | [Nest] 31  - 07/12/2024, 3:06:21 PM     LOG [ClamScanService] ClamAV is not active
```

Reply:
Sorry for my late response. Everything looks fine but for some reason Caddy doesn't get started. Could you create a seperate issue for that, so we can track and debug this further?